modules/contact/src/AugmentedLagrangianContactProblem.C

```diff
@@ -0,0 +1,174 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "AugmentedLagrangianContactProblem.h"
+
+// MOOSE includes
+#include "AuxiliarySystem.h"
+#include "DisplacedProblem.h"
+#include "MooseApp.h"
+#include "MooseMesh.h"
+#include "MooseVariable.h"
+#include "NearestNodeLocator.h"
+#include "NonlinearSystem.h"
+#include "PenetrationLocator.h"
+
+#include "SystemBase.h"
+#include "Assembly.h"
+#include "Executioner.h"
+#include "AddVariableAction.h"
+#include "ConstraintWarehouse.h"
+
+// libMesh includes
+#include "libmesh/string_to_enum.h"
+#include "libmesh/sparse_matrix.h"
+
+#include <limits>
+
+template <>
+InputParameters
+validParams<AugmentedLagrangianContactProblem>()
+{
+  InputParameters params = validParams<ReferenceResidualProblem>();
+  params.addParam<int>("maximum_lagrangian_update_iterations",
+                       100,
+                       "Maximum number of update Lagrangian Multiplier iterations per step");
+  params.addParam<Real>("penetration_tolerance", 1e-9, "The tolerance of the distance function");
```

Comment:
I think it would be better for these parameters to be defined for the individual contact constraints (penetration_tolerance, tangential_increment_stick_tolerance, tangential_friction_force_tolerance). I think I'd change their names too:
penetration_tolerance -> al_penetration_tolerance
tangential_increment_stick_tolerance -> al_incremental_slip_tolerance
tangential_friction_force_tolerance -> al_frictional_force_tolerance
They also need better descriptions.